http-client/src/main/java/io/micronaut/http/client/netty/ConnectionManager.java

```diff
@@ -0,0 +1,1263 @@
+                TimeUnit.MILLISECONDS
+            );
+            try {
+                future.await(shutdownTimeout.toMillis());
+            } catch (InterruptedException e) {
+                // ignore
+            }
+        }
+    }
+
+    public boolean isRunning() {
+        return !group.isShutdown();
+    }
+
+    public Scheduler getEventLoopScheduler() {
+        return Schedulers.fromExecutor(group);
+    }
+
+    /**
+     * Creates an initial connection to the given remote host.
+     *
+     * @param requestKey
+     * @param isStream        Is the connection a stream connection
+     * @param isProxy         Is this a streaming proxy
+     * @param acceptsEvents
+     * @param contextConsumer The logic to run once the channel is configured correctly
+     * @return A ChannelFuture
+     * @throws HttpClientException If the URI is invalid
+     */
+    private ChannelFuture doConnect(
+        DefaultHttpClient.RequestKey requestKey,
+        boolean isStream,
+        boolean isProxy,
+        boolean acceptsEvents,
+        Consumer<ChannelHandlerContext> contextConsumer) throws HttpClientException {
+
+        SslContext sslCtx = buildSslContext(requestKey);
+        String host = requestKey.getHost();
+        int port = requestKey.getPort();
+        Bootstrap localBootstrap = bootstrap.clone();
+        initBootstrapForProxy(localBootstrap, sslCtx != null, host, port);
+        localBootstrap.handler(new HttpClientInitializer(
+            sslCtx,
+            host,
+            port,
+            isStream,
+            isProxy,
+            acceptsEvents,
+            contextConsumer)
+        );
+        return localBootstrap.connect(host, port);
+    }
+
+    private void initBootstrapForProxy(Bootstrap localBootstrap, boolean sslCtx, String host, int port) {
+        Proxy proxy = configuration.resolveProxy(sslCtx, host, port);
+        if (proxy.type() != Proxy.Type.DIRECT) {
+            localBootstrap.resolver(NoopAddressResolverGroup.INSTANCE);
+        }
+    }
+
+    /**
+     * Builds an {@link SslContext} for the given URI if necessary.
+     *
+     * @return The {@link SslContext} instance
+     */
+    private SslContext buildSslContext(DefaultHttpClient.RequestKey requestKey) {
+        final SslContext sslCtx;
+        if (requestKey.isSecure()) {
+            sslCtx = sslContext;
+            //Allow https requests to be sent if SSL is disabled but a proxy is present
+            if (sslCtx == null && !configuration.getProxyAddress().isPresent()) {
+                throw customizeException(new HttpClientException("Cannot send HTTPS request. SSL is disabled"));
+            }
+        } else {
+            sslCtx = null;
+        }
+        return sslCtx;
+    }
+
+    private PoolHandle mockPoolHandle(Channel channel) {
+        return new PoolHandle(null, channel);
+    }
+
+    Mono<PoolHandle> connectForExchange(DefaultHttpClient.RequestKey requestKey, boolean multipart, boolean acceptEvents) {
```

Comment:
it would be good if you could javadoc each `connect*` method if only so other maintainers can understand the differences in behaviour in each method